starcheck/pcad_att_check.py

```diff
@@ -139,7 +139,42 @@ def make_pcad_attitude_check_report(backstop_file, or_list_file=None, attitude_f
     # gives the history of updates as a dict with a `value` and `date` key.
     sc = hopper.run_cmds(backstop_file, or_list, ofls_characteristics_file,
                          initial_state=initial_state, starcheck=True)
-    # Iterate through checks by obsid to print status
+
+    # Make maneuver structure
+    mm = []
+    for m in sc.maneuvers:
+        q1 = Quat(q=Quaternion.normalize(
+            [m['initial']['q1'], m['initial']['q2'],
+             m['initial']['q3'], m['initial']['q4']]))
+        q2 = Quat(q=Quaternion.normalize(
+            [m['final']['q1'], m['final']['q2'],
+             m['final']['q3'], m['final']['q4']]))
+
+        angle = np.degrees(2 * np.arccos(q1.q.dot(q2.q)))
```

Comment:
One thing I learned in quick functional test output review is that the maneuver angle we have been using from the matlab tools is not the great circle angle (sphere_dist) and is instead more accurately reflected as this (not sure what you call this angle - rotation matrix angle?).  I don't know if that is actually interesting at all for acquisition probability.

Reply:
For this maneyver from DEC2622A, the maneuver from 44926 to 44925 has a maneuver 'angle' of 165.05 degrees, but a sphere_dist of 14.6 deg.

```
OBSID: 44926  
RA, Dec, Roll (deg):    93.000000    23.000000   271.874615
Dither: ON Y_amp= 8.0  Z_amp= 8.0  Y_period=1000.0  Z_period= 707.1 
BACKSTOP GUIDE_SUMM OR DOT TLR 

MP_TARGQUAT at 2023:001:10:27:59.567 (VCDU count = 7072498)
  Q1,Q2,Q3,Q4: -0.36518414  -0.59294570  0.41533499  0.58528010
  MANVR: Angle=  74.61 deg  Duration= 1655 sec  Slew err= 51.3 arcsec  End= 2023:001:10:55:40

==================================================================================== 
OBSID: 44925  
RA, Dec, Roll (deg):   108.597458    20.909120    70.938652
Dither: ON Y_amp= 8.0  Z_amp= 8.0  Y_period=1000.0  Z_period= 707.1 
BACKSTOP GUIDE_SUMM OR DOT TLR 

MP_TARGQUAT at 2023:001:11:50:04.357 (VCDU count = 7091716)
  Q1,Q2,Q3,Q4: 0.45300842  0.37715544  0.71183683  0.38187101
  MANVR: Angle= 165.05 deg  Duration= 2861 sec  Slew err= 27.6 arcsec  End= 2023:001:12:37:50

```

Reply:
This is a great point and one I hope we'll remember. Fundamentally the way I would think about this is:
```
dq = q1.dq(q2)  # quat getting from q1 to q2 (or vice versa, I never remember)
Then dq.q[3] = cos(2 * alpha) where alpha is the rotation angle about q[0:2].
I'm assuming that q1.q .dot (q2.q) is the same as alpha?

Q56. How do I convert a rotation axis and angle to a quaternion?
----------------------------------------------------------------

  Given the rotation axis and angle, the following algorithm may be
  used to generate a quaternion:

    ------------------------------------------------
    sin_a = sin( angle / 2 )
    cos_a = cos( angle / 2 )

    q -> x    = axis -> x * sin_a
    q -> y    = axis -> y * sin_a
    q -> z    = axis -> z * sin_a
    q -> w    = cos_a

    quaternion_normalise( q );
    ------------------------------------------------
```
I almost wonder if we should add a method to `Quat` like `Quat.maneuver_angle(self, q2)`.